### PR TITLE
Fix issue where schema.xml etc is put in a different dir than the cor…

### PR DESCRIPTION
**Pull request by jakedaleweb**
…e is instantiated in

### CODE REVIEW

**Comment by jakedaleweb** [comment]
What this was doing was POSTing schema.xml to
`/my/solr/location/silverstripe-namespaced-myindex/`

And the core would then be instantiated in 
`/my/solr/location/myindex/`
Without any of the necessary config files, this would in turn cause the Configure to fail.

**Comment by dhensby** [comment]
I've noticed that you've created this pull request from one of the "core branches" rather than a branch specifically intended for this pull request. This can cause problems if we ask you to target this PR at a different branch. No action is needed at the moment, this is for your information.

In the future please create a new branch for each individual pull request. See our [guide about contributing PRs](https://docs.silverstripe.org/en/4/contributing/code/#step-by-step-from-forking-to-sending-the-pull-request)

**Comment by dhensby** [comment]
@andrewandante tested this locally with the current version of the module and was unable to replicate the described problem...

Has this fix been tested "in the wild" to confirm it is indeed the source of the problem?

**Comment by jakedaleweb** [comment]
@dhensby issue is occurring on a client site and the fix has been tested there.

The work up to this point fixes the Configure task but the Reindex task is now broken (as it relies on the non-namespaced path name) - I am looking in to this now.

**Comment by jakedaleweb** [comment]
Reindex issue has been resolved in `serviceForCore()`

**Comment by jakedaleweb** [comment]
To be clear, at this stage, the module now uses the fully name spaced name across the board for all calls to solr, there were problems with consistency around this earlier where:
- schema.xml etc were being sent to the non-name spaced directory
- the actual cores were instantiated with name spaced dirs
- reindex would attempt to send data to the non name spaced dirs

**Comment by jakedaleweb** [comment]
This branch is now running on mediumtest-uat if you would like to see it in action, the broken version is running on training-uat.

I'm aware there is a broken test now, I'm looking in to it.

File: src/Solr/SolrIndex.php [hunk 1]
```diff
@@ -124,29 +124,6 @@ public function getIndexName()
         return implode($indexParts);
     }
 
-    /**
-     * Helper for returning the indexer class name from an index name, encoded via {@link getIndexName()}
-     *
-     * @param string $indexName
-     * @return string
-     */
-    public static function getClassNameFromIndex($indexName)
```
Comment (jakedaleweb) [inline src/Solr/SolrIndex.php]: `getIndexName()` in this class serves an almost identical purpose.

File: tests/SolrIndexTest.php [hunk 2]
```diff
@@ -376,30 +376,6 @@ public function testGetIndexNameWithPrefixAndSuffixFromEnvironment()
         );
     }
 
-    /**
-     * @dataProvider indexNameProvider
-     * @param string $indexName
-     * @param string $expected
-     */
-    public function testGetClassNameFromIndex($indexName, $expected)
```
Comment (jakedaleweb) [inline tests/SolrIndexTest.php]: Method `getClassNameFromIndex()` was removed, hence the test is also removed

File: src/Solr/SolrIndex.php [hunk 3]
```diff
@@ -1051,17 +1052,20 @@ public function setService(SolrService $service)
      */
     public function uploadConfig($store)
     {
+        // Ensure the short class name is used to match the call which creates the core in SolrService->coreCreate()
+        $indexName = self::getClassNameFromIndex($this->getIndexName());
+        $indexName = ClassInfo::shortName($indexName);
```
[the remark this answers is not in the document]
Comment (jakedaleweb) [reply src/Solr/SolrIndex.php]: PR has been changed as such

**Comment by robbieaverill** [comment]
Summarising a quick Slack convo with @jakedaleweb about this:

Remaining error: `    "msg": "undefined field _locale",`

FluentSearchVariant (in tractorcow/silverstripe-fluent) applies itself to the environment if the extension is applies, but causes this error on queries when `Locale::getCached()->count();` returns 0.

I fixed this in SS3 for fulltextsearch recently in https://github.com/silverstripe/silverstripe-fulltextsearch/pull/202

Until merge up/for now we can and should also fix this in FluentSearchVariant by also checking that `Locale::getCached()->count() > 0` in appliesToEnvironment

**Comment by jakedaleweb** [comment]
@robbieaverill Referenced PR is here https://github.com/tractorcow/silverstripe-fluent/pull/372/files

This is running on mediumtest-uat, we have successful search there 🙆

**Pull request merged**